### PR TITLE
DOC-9626: compatibility caveats and tweaks

### PR DESCRIPTION
**Pull request by osfameron**
Added notes from https://issues.couchbase.com/browse/JSCBC-978 about omitted features, and edited appropriate doc pages where I could find them.

Few mentions of "Couchbase++" which are uncertain (@AV25242 and @raycardillo on the case)

### CODE REVIEW

File: modules/concept-docs/pages/durability-replication-failure-considerations.adoc [hunk 1]
```diff
@@ -16,7 +16,14 @@ include::{version-server}@sdk:shared:partial$durability-replication-failure-cons
 include::{version-server}@sdk:shared:partial$durability-replication-failure-considerations.adoc[tag=syncrep]
 include::{version-server}@sdk:shared:partial$durability-replication-failure-considerations.adoc[tag=syncrep3]
 
-include::{version-server}@sdk:shared:partial$durability-replication-failure-considerations.adoc[tag=older]
+== Older Server Versions
+// include::{version-server}@sdk:shared:partial$durability-replication-failure-considerations.adoc[tag=older]
+If a version of Couchbase Server lower than 6.5 is being used then the fallback is 'client verified' durability.
+
+WARN: Client Verified durability is supported in xref:3.2@durability-replication-failure-considerations.adoc#older-server-versions[Node.js SDK 3.2] but not in 4.0.
```
Comment (RichardSmedley) [inline modules/concept-docs/pages/durability-replication-failure-considerations.adoc]: Shouldn't this be WARNING ?
https://docs.antora.org/antora/latest/asciidoc/admonitions/

**Review by RichardSmedley — COMMENTED** [review]
:+1:

**Review by maria-robobug — APPROVED** [review]
👍 LGTM

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc [hunk 2]
```diff
@@ -524,11 +524,26 @@ Learn more in the xref:howtos:encrypting-using-sdk.adoc#migrating-from-2[encrypt
...
+* Replica reads 
+
+In addition:
+
 * The required minimum version of Node.js is now v12 (the oldest supported maintenance LTS release).
 * Ping error entries are now typed.
-// view query reduce option string-> boolean // TODO word this more clearly
-* Search scan consistency `request-plus` has been removed.
 * `get` requests on locked documents now retry rather than fast-fail.
+
```
Comment (maria-robobug) [inline modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc]: Probably don't need all these lines.

Comment (osfameron) [reply modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc]: What if we need to add anything else!!!! 😂

**Comment by osfameron** [comment]
Keeping gh-pages up-to-date: https://osfameron.github.io/DOC-9626-NodeJS-4-migration/nodejs-sdk/current/hello-world/start-using-sdk.html

File: modules/howtos/pages/slow-operations-logging.adoc [hunk 3]
```diff
@@ -7,8 +7,11 @@
 {description}
 Change the settings to alter how much information you collect
 
-To improve debuggability certain metrics are automatically measured and logged.
-These include slow queries, responses taking beyond a certain threshold, and orphanned responses.
+WARNING: the Logging implementation has changed substantially in 4.0 and is not currently fully documented.
+This will be resolved in a future 4.x release.
+
+To improve debugability certain metrics are automatically measured and logged.
```
Comment (programmatix) [inline modules/howtos/pages/slow-operations-logging.adoc]: debuggability

Comment (osfameron) [reply modules/howtos/pages/slow-operations-logging.adoc]: Hah, both are quite yucky 🤢 
I have conferred with my esteemed Style and Tone committee, who have found in your favour, will revert 👍

**Review by RichardSmedley — APPROVED** [review]
:+1:

**Review by ingenthr — APPROVED** [review]
Had a good look over this and I think it's good @maria-robobug .

**Pull request merged**